runtime/vm/resolvesupport.cpp

```diff
@@ -80,6 +82,18 @@ finalFieldSetAllowed(J9VMThread *currentThread, bool isStatic, J9Method *method,
 				}
 				legal = false;
 			}
+
```

Comment:
Would it be better to check for record first? I think there may be a possibility of setting the exception twice here.
```
ifdef
if (record) {
   your new code
} else
endif
...old code...

Reply:
Also, why is the extra check required at all? All modern VMs disallow setting of final fields from outside of constructors.

Reply:
Yes I think you are right.

Reply:
I retested and it looks like the jcl addition alone is sufficient.

Reply:
updated.